src/Logic.hs

```diff
@@ -251,57 +251,63 @@ updateStateVar var state = void $ atomically $ swapTMVar var state
 readStateVar :: StateVar -> IO ProjectState
 readStateVar var = atomically $ readTMVar var
 
+clearPullRequest :: PullRequestId -> PullRequest -> ProjectState -> Action ProjectState
+-- Closes and opens a new PR with the same id. Useful for clearing approval and build status safely.
+clearPullRequest prId pr state =
+  let
+    branch = Pr.branch pr
+    title  = Pr.title pr
+    author = Pr.author pr
+    baseBranch = Pr.baseBranch pr
+    sha    = Pr.sha pr
+  in
+    handlePullRequestClosed prId state >>= 
```

Comment:
It would leave a comment if the PR is the integration candidate. In that case it will also clear the integration candidate. 

The "Abandoning this pull request because it was closed." message is not appropriate though. I think fixing issue #66 should display an appropriate message for this case as well.

Reply:
This change does introduce the same bug in a new location, though.

By substituting the closed/opened function definitions, you can leave out the parts that are wrong:

```haskell
    Pr.insertPullRequest prId branch baseBranch sha title author . Pr.deletePullRequest prId <$>
      case Pr.integrationCandidate state of
        Just candidatePr | candidatePr == prId ->
          pure state { Pr.integrationCandidate = Nothing }
        _notCandidatePr -> pure state
```

That should then also fix #66, right?

Reply:
Yes, this should fix it. But I think it's useful if hoff shows a message that it stopped the integration. We can use different messages for different cases.

Reply:
Oh wait, no, that would remove all the comments. In that case, you can just change the comment to `Abandoning pull request because it changed after approval`. That would cover both having commits and the base branch changed :+1:

Reply:
But if we changed the comment completely, we would get an incorrect message when a user actually closes a PR. I think maybe we should change the message depending on the situation. I guess it would be useful to know what caused the abandoning.

Reply:
Oh, I guess you are talking about inlining the `handlePullRequestClosed` in `clearPullRequest`. I guess that's going to work. I still think showing different messages would be nice, but idk...

Reply:
Yeah, indeed. For closed pull requests it would still call `handlePullRequestClosed` with the correct message. In here the message is indeed generic over rebase and commits pushed, but I think that's fine. The change can be seen in the PR right above the comment that would be posted.

Reply:
That's true, good point.

Reply:
I suggest implementing this in another PR. I'll need to add a test for when a commit is made as well and I am afraid this PR will grow outside of its scope. Wdyt?

Reply:
Alright! Good call on limiting the scope :+1: